digital_land/collection.py

```diff
@@ -555,3 +557,45 @@ def entry_date(self, entry):
         return entry.get(
             "entry-date", datetime.utcnow().strftime("%Y-%m-%dT%H:%H:%M:%SZ")
         )
+
+    def retire_endpoints_and_sources(collection, collection_df_to_retire) -> None:
+        """
+        This method should be called from commands.py with the correct input.csv format per unique collection.
+
+        Args:
+            collection (Collection): The Collection object which gets instantiated by a corresponding commands.py function which calls this method.
+            collection_df_to_retire (str): A small dataframe of sources and endpoints to be retired within a colleciton, this should be
+                generated by a function call within commands.py.
+        """
+
+        try:
+            endpoint_csv_path = os.path.join(collection.dir, "endpoint.csv")
+            source_csv_path = os.path.join(collection.dir, "source.csv")
+
+            # Read endpoint and source CSV files
+            endpoint_csv_df = pd.read_csv(endpoint_csv_path)
+            source_csv_df = pd.read_csv(source_csv_path)
+
+            # Get today's date in the format YYYY-MM-DD
+            today_date = datetime.now().strftime("%Y-%m-%d")
+
+            # Update end_date column in endpoint file
+            endpoint_csv_df.loc[
+                endpoint_csv_df["endpoint"].isin(collection_df_to_retire["endpoint"]),
+                "end-date",
+            ] = today_date
+
+            # Update end_date column in source file
+            source_csv_df.loc[
+                source_csv_df["source"].isin(collection_df_to_retire["source"]),
+                "end-date",
+            ] = today_date
```

Comment:
Would it be possible to display the sources/endpoints that have been ended? Or even just log an error for the sources/endpoints that weren't found in the endpoint.csv and source.csv files? As someone may run this with a whole batch of sources, one or two of them have a mistake in the hash and not found but they'd currently have no idea

Reply:
This is a very good point, I'll look to modifying this

Reply:
@CarlosCoelhoSL I've adjusted the function to fail completely if any source/endpoint hashes have no match found, and then print these values for reference